ui/ui-components/client/client.ts

```diff
@@ -195,15 +142,85 @@ export class Client {
           error.response?.data?.error ??
           error;
 
-        options.errorHandler.set({ message: err });
-        const newError = Error(err);
+        const newError = new Error(err);
         newError["code"] = error.response?.data?.error?.code;
         throw newError;
       } else {
-        options.errorHandler.set({ message: error });
         throw error;
       }
     }
+  }
+
+  public request = async <Response = any>(
+    verb: Method = "get",
+    path: string,
+    data: AxiosRequestConfig["data"] = {},
+    options: Partial<ClientRequestOptions> = {}
+  ): Promise<Response> => {
+    options = { ...Client.defaultOptions, ...options };
+    const headers: AxiosRequestHeaders = { ...Client.defaultHeaders };
+
+    try {
+      const { req } = this.getRequestContext();
+      if (req?.headers?.cookie) {
+        headers["X-GROUPAROO-SERVER-TOKEN"] = this.serverToken;
+        headers["cookie"] = req?.headers?.cookie;
+        options.useCache = false; // do not ever cache responses on the server
+      }
+
+      const config: AxiosRequestConfig = {
+        url: `${this.webUrl}/api/${this.apiVersion}${path}`, // path comes with a leading "/"
+        withCredentials: true,
+        method: verb.toLowerCase() as Method,
+        headers,
+      };
+
+      data.csrfToken = this.csrfToken();
+
+      if (data) {
+        for (const i in data) {
+          if (data[i] === null || data[i] === undefined) {
+            delete data[i];
+          }
+        }
+
+        if (data.file || data._file) {
+          delete config.headers;
+          let dataForm = new FormData();
+          for (const i in data) {
+            dataForm.append(i, data[i]);
+          }
+          data = dataForm;
+        }
+
+        if (config.method === "get") {
+          config.params = data;
+        } else {
+          config.data = data;
+        }
+      }
+
+      let unlock: ClientCacheGetObject["unlock"];
+      if (config.method === "get" && options.useCache) {
+        const { cacheData, unlock: _unlock } = await this.cache.get<Response>(
+          config.url + JSON.stringify(data)
+        );
+        if (cacheData) {
+          return cacheData;
+        }
+        unlock = _unlock;
+      } else {
+        this.cache.clear();
+      }
+
+      return await this.sendRequest(config, unlock);
+    } catch (error) {
+      if (options.errorHandler) {
+        options.errorHandler.set({ message: error });
+        return {} as Response;
```

Comment:
Confirming that returning an empty object was what the old useApi hook used to do:

https://github.com/grouparoo/grouparoo/blob/161da0ce3daeade174361526551f2a40c6cc490b/ui/ui-components/hooks/useApi.ts#L58-L59

Reply:
Yep, this is the same as it was before